### PR TITLE
fix(gateway): Use Deno's WebSocket

### PR DESCRIPTION
**Pull request by Endy3032**
Check for Deno namespace and use its ws instead of node ws (without using eval like #3029)

### CODE REVIEW

File: packages/gateway/src/Shard.ts [hunk 1]
```diff
@@ -129,13 +133,19 @@ export class DiscordenoShard {
     url.searchParams.set('v', this.gatewayConfig.version.toString())
     url.searchParams.set('encoding', 'json')
 
-    const socket = new WebSocket(url.toString())
+    let socket: NodeWebSocket
+
+    // @ts-expect-error Deno
+    if (isDeno) socket = new WebSocket(url.toString())
+    else if (isNode) socket = new NodeWebSocket(url.toString())
+    else throw new Error('Unsupported runtime')
```
Comment (Skillz4Killz) [inline packages/gateway/src/Shard.ts:141]: Should use node still. We do support other runtimes like bun which support ws. Its just deno that doesnt support ws

Comment (Skillz4Killz) [reply packages/gateway/src/Shard.ts:141]: Just default to node if not deno like socket = isDeno ? Websocket : Nodewebsocket

**Review by Skillz4Killz — APPROVED** [review]

File: packages/gateway/src/Shard.ts (hunk 1 above)
Comment (Endy3032) [reply packages/gateway/src/Shard.ts:141]: This was what ITOH suggested to me in DM since he said bun is not fully compatible yet

Comment (Skillz4Killz) [reply packages/gateway/src/Shard.ts:141]: im pretty sure bun works with ws but if it doesnt we need to find a lib for it. @Tricked-dev any thoughts?

**Review by Skillz4Killz — APPROVED** [review]

**Pull request merged**